gboard_autopsy.py

```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
+
+        Args:
+            input_dir (str): Input directory
+
+        Returns:
+            A dict python object representing the JSON output reported by the
+            tool.
+        """
+
+        self.log(Level.INFO, "Running analyzer on " + input_dir + " folder")
+
+        cmd_args = self.generate_analyzer_args(input_dir)
+        process = subprocess.Popen(cmd_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        cmd_out, cmd_err = process.communicate()
+        if process.returncode != 0:
+            raise IngestModuleException('Executable failed to execute! \nstdout:' + cmd_out + '\nstderr:' + cmd_err)
+
+        return json.loads(cmd_out)
+
+    def generate_analyzer_args(self, input_dir):
+        return [
+            self.path_to_exe,
+            '-d', input_dir,
+        ]
+
+    def get_input_dir(self, data_source, file_manager):
+        files = file_manager.findFiles(data_source, "%" + self.GBOARD_PACKAGE_NAME + "%")
+
+        for file in files:
+            if file.isFile() == True:
+                path = file.getLocalAbsPath()
+                if path == None:
+                    continue
+
+                return self.find_first_folder(path, self.GBOARD_PACKAGE_NAME)
+
+        raise IngestModuleException('Folder ' + self.GBOARD_PACKAGE_NAME + ' not found in ' + data_source.getName() + ' data source!')
+
+    def find_first_folder(self, path, folder):
+        normalized_path = os.path.normpath(path)
+        splitted_path = normalized_path.split(os.path.sep)
+
+        for idx, dir in enumerate(splitted_path):
+            if dir == folder:
+                return os.path.sep.join(splitted_path[:idx+1])
+
+        # Not found
+        return None
+
+    def report_analysis(self, input_dir, data_source, blackboard, file_manager, analysis_output):
+        for dictionary in analysis_output['dictionaries']:
+            file = self.get_common_sufix_file(data_source, file_manager, input_dir, dictionary['path'])
+            for entry in dictionary['entries']:
+                self.publish_analysis_artifact(blackboard, file, self.dictionary_art_type, [
+                    (self.dictionary_word_attr_type, entry['word']),
+                    (self.dictionary_shortcut_attr_type, entry['shortcut']),
+                    (self.dictionary_locale_attr_type, entry['locale'])
+                ])
+
+    def get_common_sufix_file(self, data_source, file_manager, common_path, full_path):
+        if not common_path.endswith(os.path.sep):
+            sanitized_path = common_path + os.path.sep
+        else:
+            sanitized_path = common_path
+
+        rel_path = full_path.split(sanitized_path)
+        if len(rel_path) > 1:
+            full_gboard_path = os.path.join(self.GBOARD_PACKAGE_NAME, os.path.dirname(rel_path[1]))
+            files = file_manager.findFiles(data_source, os.path.basename(rel_path[1]), full_gboard_path)
+            return files[0] if len(files) else None
```

Comment:
You can replace `len(files)` with `files`

Reply:
Done